src/Main.elm

```diff
@@ -1,37 +1,143 @@
-module Main exposing (Model, Msg(..), book, init, main, update, view)
+module Main exposing (Model, Msg(..), init, main, update, view)
 
 import BookData exposing (Book, allBooks)
 import Browser
-import Html exposing (Html, div, h1, img, li, text, ul)
+import Html exposing (Html, button, div, h1, img, li, table, td, text, th, thead, tr, ul)
 import Html.Attributes exposing (alt, class, src)
+import Html.Events exposing (onClick)
+import List exposing (append, filter, map, sum)
 
 
 
 ---- MODEL ----
 
 
 type alias Model =
-    { books : List Book }
+    { books : List Book, cart : Cart }
+
+
+type alias CartItem =
+    { id : BookID, quantity : Int }
+
+
+type alias Cart =
+    List CartItem
+
+
+type alias BookID =
+    Int
+
+
+add : Cart -> BookID -> Cart
+add cart bookID =
+    -- let
+    --     _ =
+    --         Debug.log "Logging b" bookID
+    -- in
+    case inCart bookID cart of
+        True ->
+            let
+                incrementQuantity cartItem =
+                    if cartItem.id == bookID then
+                        { cartItem | quantity = cartItem.quantity + 1 }
+
+                    else
+                        cartItem
+            in
+            List.map (\item -> incrementQuantity item) cart
+
+        False ->
+            append cart [ { id = bookID, quantity = 1 } ]
+
+
+remove : Cart -> BookID -> Cart
+remove cart bookID =
+    List.filter (\cartItem -> cartItem.id /= bookID) cart
+
+
+increment : Cart -> BookID -> Cart
+increment cart bookID =
+    let
+        incrementQuantity cartItem =
+            if cartItem.id == bookID then
+                { cartItem | quantity = cartItem.quantity + 1 }
+
+            else
+                cartItem
+    in
+    List.map (\item -> incrementQuantity item) cart
+
+
+decrement : Cart -> BookID -> Cart
+decrement cart bookID =
+    let
+        decrementQuantity cartItem =
+            if cartItem.id == bookID then
+                { cartItem | quantity = cartItem.quantity - 1 }
+
+            else
+                cartItem
+    in
+    List.map (\item -> decrementQuantity item) cart
 
 
 init : ( Model, Cmd Msg )
 init =
-    ( { books = allBooks }
+    ( { books = allBooks, cart = [] }
     , Cmd.none
     )
 
 
+getQuantity : Cart -> BookID -> Int
+getQuantity cart bookID =
```

Comment:
I think this could simply be called `quantity`. However, now I'm wondering whether a different approach might be better. Maybe, instead of having a list of books + a cart, you could just have the list of books, with another field for quantity? And then everything with a quantity of >0 is in the cart ...  
Especially with seeing your other getter functions below, I think this might be a better approach. Sorry, I steered you towards storing a list of books and a cart.